python/mepipelineappintg/metadetect_pizza_cutter_tools.py

```diff
@@ -38,6 +39,79 @@ def _execute_query(query, dbh, verbose, Timing):
     return rowds
 
 
+@lru_cache(maxsize=1024)
+def _do_piff_info_query(
+    dbh, verbose, Timing, expnum, PiffTag, relPrefix, dbSchema
+):
+    query = f"""SELECT
+                    DISTINCT
+                    qa.filename,
+                    qa.flag as desdm_flags,
+                    qa.fwhm_cen,
+                    qa.star_t_std,
+                    qa.star_t_mean,
+                    qa.nstar,
+                    qa.exp_star_t_mean,
+                    qa.exp_star_t_std
+                FROM
+                    {dbSchema:s}{relPrefix:s}proctag t,
+                    {dbSchema:s}{relPrefix:s}miscfile m,
+                    {dbSchema:s}{relPrefix:s}PIFF_MODEL_QA qa
+                where
+                    qa.expnum = {expnum:d}
+                    and t.tag = '{PiffTag:s}'
+                    and t.pfw_attempt_id = m.pfw_attempt_id
+                    and m.filetype = 'piff_model'
+                    and m.filename = qa.filename
+                """
+
+    return _execute_query(query, dbh, verbose, Timing)
+
+
+def add_piff_info_to_yaml(
+    yaml_data, PiffTag, dbh, dbSchema, releasePrefix=None,
+    Timing=False, verbose=0
+):
+    """Get coadd info from attemp ID.
+
+        Inputs:
+            yaml_data: the yaml data to adjust
+            PiffTag:   the Piff tag to use
+            dbh:       Database connection to be used
+            dbSchema:  Schema over which queries will occur.
+            releasePrefix: Prefix string (including _'s) to identify a specific
+                           set of tables
+                           (Useful when working from releases in DESSCI).
+                           None --> will substitute a null string.
+            Timing:    Causes internal timing to report results.
+            verbose:   Integer setting level of verbosity when running.
+
+        Returns:
+            coadd_data: A dictionary containing the relevant coadd data.
+    """
+    if releasePrefix is None:
+        relPrefix = ""
+    else:
+        relPrefix = releasePrefix
+
+    for band in yaml_data:
+        for isrc in range(len(yaml_data[band]["src_info"])):
+            expnum = yaml_data[band]["src_info"][isrc]["expnum"]
+            piff_rows = _do_piff_info_query(
```

Comment:
@rgruendl I loop through the single-epoch images here to get everything.

Reply:
The loop does a query per expnum and then finds the right file for the CCD. I am caching the query results so that things are fast.

Reply:
Good enough for me.